lib.nix

```
@@ -0,0 +1,60 @@
+            build-static-release =
+              {
+                cargo-zigbuild,
+                rust-toolchain,
+                mimalloc,
+                git,
+                writeShellApplication,
+                extra-runtimeInputs ? [ ],
+              }:
+
+              writeShellApplication {
+                name = "build-static-release";
+                runtimeInputs = [
+                  cargo-zigbuild
+                  mimalloc
+                  rust-toolchain
+                  git
+                ] ++ extra-runtimeInputs;
+
+                text = ''
+                  if [ -n "$(git status --porcelain)" ]; then
+                    echo >&2 'dirty tree, refusing to build; stash or commit your changes'
+                    exit 1
+                  fi
+                  set -x
+                  # zig 0.14.0 causes ProcessFdQuotaExceeded on macos, so raise the limit.
+                  # this fails if some parent process has already called `ulimit -n`, so
+                  # you may need to open a new shell in that case.
+                  ${if pkgs.stdenv.isDarwin then "ulimit -n 8192" else ""} #
+                  # mimalloc uses __DATE__, __TIME__ macros
+                  export CFLAGS="-Wno-error=date-time"
```

can generalize comment here
```suggestion
                  # `zig cc` sets `-Werror=date-time` which breaks some dependencies. not needed in this case
                  # as this tool is not meant for reproducibility
                  export CFLAGS="-Wno-error=date-time"
```